chia/wallet/wallet_state_manager.py

```diff
@@ -636,14 +637,13 @@ async def new_coin_state(
     ) -> None:
         # TODO: add comment about what this method does
 
-        # Sort by created height, then add the reorg states (created_height is None) to the end
-        created_h_none: List[CoinState] = []
-        for coin_st in coin_states.copy():
-            if coin_st.created_height is None:
-                coin_states.remove(coin_st)
-                created_h_none.append(coin_st)
-        coin_states.sort(key=lambda x: x.created_height, reverse=False)  # type: ignore
-        coin_states.extend(created_h_none)
+        # Input states should already be sorted by cs_height, with reorgs at the beginning
+        curr_h = -1
+        for c_state in coin_states:
+            if last_change_height_cs(c_state) < curr_h:
+                raise ValueError("Input coin_states is not sorted properly")
+            curr_h = last_change_height_cs(c_state)
```

Comment:
Function calls aren't the cheapest thing ever for tight loops.  I don't think this particularly reduces legibility so it's probably a good change.

```suggestion
            last_change_height = last_change_height_cs(c_state)
            if last_change_height < curr_h:
                raise ValueError("Input coin_states is not sorted properly")
            curr_h = last_change_height
```